### DISCUSSION TITLE
Use !include with private URL

### DISCUSSION THREAD

**Discussion by vmanikes** [structurizr/java#287] [Q&A]
### Question

I am trying to build a parent-child relationship using C4 where parent is the root.dsl and all the children aka microservices will contain a arch.dsl in their repos. The root.dsl will contain all the microservice's arch.dsl, but the problem is that the arch.dsl will be a private repo. Can someone tell me how to use private URL's with !include?

**Comment by simonbrowndotje** [comment]
This is the recommended approach -> https://github.com/structurizr/dsl/issues/134
